### PR TITLE
🌱  Logs/update type casing

### PR DESCRIPTION
**Pull request by killianmuldoon**
Bring more of our logs into line with the logging guidelines by capitalising the Kinds in the key/value pairs.

/kind cleanup

### CODE REVIEW

**Comment by killianmuldoon** [comment]
/hold
This PR includes https://github.com/kubernetes-sigs/cluster-api/pull/7075 so we should wait for that to merge first.

File: exp/util/util.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind, log logr.Lo
 			return nil
 		}
 
-		log.V(4).Info("projecting object", "namespace", m.Namespace, "name", ref.Name)
+		log.V(4).Info("projecting object", "Namespace", m.Namespace, "Name", ref.Name)
```
Comment (killianmuldoon) [inline exp/util/util.go:82]: Question that needs an answer to ensure consistency:
Should namespace and name be capitalised? (or either or neither 🙂)

File: exp/util/util.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func GetMachinePoolByName(ctx context.Context, c client.Client, namespace, name
 func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind, log logr.Logger) handler.MapFunc {
 	log = log.WithValues("machine-pool-to-infra-map-func", gvk.String())
 	return func(o client.Object) []reconcile.Request {
-		log := log.WithValues("namespace", o.GetNamespace(), "name", o.GetName())
+		log := log.WithValues("Namespace", o.GetNamespace(), "name", o.GetName())
 		m, ok := o.(*expv1.MachinePool)
 		if !ok {
 			log.V(4).Info("not a machine pool")
```
Comment (sbueringer) [inline exp/util/util.go:69]:
```suggestion
			log.V(4).Info("not a machine pool", "Object", klog.KObj(o))
```
\+ drop l.66

> When Kubernetes object kind is unknown without runtime checking we should use object key.
> https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments

Comment (sbueringer) [reply exp/util/util.go:69]: \+ Add 
```go
log := log.WithValues("MachinePool", klog.KObj(o))
```
after the if

and drop namespace/name from l.82

File: internal/controllers/machinedeployment/machinedeployment_sync.go [hunk 3]
```diff
@@ -236,13 +237,13 @@ func (r *Reconciler) getNewMachineSet(ctx context.Context, d *clusterv1.MachineD
...
 	case err != nil:
-		log.Error(err, "Failed to create new machine set", "machineset", newMS.Name)
+		log.Error(err, "Failed to create new MachineSet", "MachineSet", klog.KObj(&newMS))
 		r.recorder.Eventf(d, corev1.EventTypeWarning, "FailedCreate", "Failed to create MachineSet %q: %v", newMS.Name, err)
 		return nil, err
 	}
 
 	if !alreadyExists {
-		log.V(4).Info("Created new machine set", "machineset", createdMS.Name)
+		log.V(4).Info("Created new MachineSet", "MachineSet", klog.KObj(&newMS))
```
Comment (sbueringer) [inline internal/controllers/machinedeployment/machinedeployment_sync.go:246]:
```suggestion
		log.V(4).Info("Created new MachineSet", "MachineSet", klog.KObj(&createdMS))
```
Probably (?)

File: exp/util/util.go (hunk 1 above)
Comment (sbueringer) [reply exp/util/util.go:82]: Made another suggestion. I think we can get rid of this entirely

**Review by sbueringer — COMMENTED** [review]
Thank you very much! :)

A few nits. The underlying PR should merge soon

File: exp/util/util.go (hunk 2 above)
Comment (sbueringer) [reply exp/util/util.go:69]: Regarding the first log message. Maybe we can just drop it. Looked through a bunch of other handler.Funcs and we usually just return nil without log, e.g.:
```go
		cluster, ok := o.(*clusterv1.Cluster)
		if !ok {
			return nil
		}
```

Comment (killianmuldoon) [reply exp/util/util.go:69]: I'm going to leave the log there for now, but I've made the other changes.

File: internal/controllers/machinedeployment/machinedeployment_sync.go (hunk 3 above)
Comment (killianmuldoon) [reply internal/controllers/machinedeployment/machinedeployment_sync.go:246]: It's the same at this point, right? I'll update as it might be more clear.

Comment (sbueringer) [reply internal/controllers/machinedeployment/machinedeployment_sync.go]: Hm yeah, good point. Just saw the diff to before

**Comment by sbueringer** [comment]
/lgtm

cc @ykakarap

**Review by ykakarap — COMMENTED** [review]
Just nits.

**Comment by fabriziopandini** [comment]
lgtm pending last round of nits

**Pull request merged**